### PR TITLE
return 0 in getCriticalSectionSize for wasm

### PR DESCRIPTION
**Pull request by skoppe**
https://github.com/ldc-developers/ldc/issues/3118

### CODE REVIEW

File: gen/target.cpp [hunk 1]
```diff
@@ -78,6 +78,9 @@ unsigned getCriticalSectionSize(const Param &params) {
     break;
   }
 
+  if (arch == llvm::Triple::wasm32 || arch == llvm::Triple::wasm64)
+    return 0;
+
```
Comment (kinke) [inline gen/target.cpp]: Thx! - Please get rid of the changes in the other files (part of the front-end [except for the init in main.cpp], where LDC-specifics are guarded by `version (IN_LLVM)` to simplify future front-end merges - and DMD won't feature wasm support anytime soon, if ever).

File: dmd/target.d [hunk 2]
```diff
@@ -321,6 +321,10 @@ else // !IN_LLVM
             // sizeof(pthread_mutex_t) for Solaris.
             return 24;
         }
+        else if (params.isWasm)
+        {
+            return 0;
+        }
```
Comment (kinke) [inline dmd/target.d]: [This is the upstream implementation of `getCriticalSectionSize()` and excluded from LDC, see line 123. `Target` is a front-end class which heavily depends on the back-end, that's why LDC only uses a few functions from upstream.]

File: gen/target.cpp (hunk 1 above)
Comment (skoppe) [reply gen/target.cpp]: you'll still need the changed to `dmd/globals.h` right? else where does `global.params.isWasm` point to in main.cpp:init?

Comment (kinke) [reply gen/target.cpp]: I meant `main.cpp` being the only other file outside the front-end, but the change in there is superfluous, we just need the 3 lines above - and ideally a test, maybe adding a `-w` here? https://github.com/ldc-developers/ldc/blob/d04e7081511a5d14546ab8eec6f3b66c13d62ec5/tests/baremetal/wasm.d#L5

Comment (skoppe) [reply gen/target.cpp]: done and done

**Pull request merged**